src/ui/MessageItemMenu/index.tsx

```diff
@@ -209,9 +211,13 @@ export default function MessageItemMenu({
                 <MenuItem
                   className="sendbird-message-item-menu__list__menu-item menu-item-delete"
                   onClick={() => {
-                    if (isMultipleFilesMessage(message) || !disabled) {
-                      showRemove(true);
-                      closeDropdown();
+                    if (!disabled || isFailedMessage(message)) {
+                      if (isFailedMessage(message)) {
+                        channelStore?.deleteMessage?.(message);
+                      } else {
+                        showRemove(true);
+                        closeDropdown();
+                      }
```

Comment:
Instead of using nested if stmt and repeat the same condition `isFiledMessage`, we can just do the early return like below 

```suggestion
                    if (disabled) {
                      return;
                    }
                    if (isFailedMessage(message)) {
                      channelStore?.deleteMessage?.(message);
                    } else {
                      showRemove(true);
                      closeDropdown();
                    }
```

Reply:
nice!